### ISSUE TITLE
When running on App Service the wrong client IP is reported

### ISSUE THREAD

**Issue by derkoe** [microsoft/ApplicationInsights-Java#1805]
### Expected behavior
We are using the App Insights Java agent on App Service and have [IP masking turned off](https://docs.microsoft.com/en-us/azure/azure-monitor/app/ip-collection?tabs=net#storing-ip-address-data). The expected behavior is that "request" logs should include the correct client IP in the field `client_IP`.

### Actual behavior
The field `client_IP` is filled with one of the outbound IP addresses of the app service.

### To Reproduce
* Deploy an App Service with Java 11 on Linux and App Inisights enabled
* [Turn off IP masking for the App Insights](https://docs.microsoft.com/en-us/azure/azure-monitor/app/ip-collection?tabs=net#storing-ip-address-data)
* Deploy any Spring Boot app (you can use https://github.com/derkoe/snoop) as Java JAR application
* Make some requests
* Check the field `client_IP` in the App Insights log type "request"

#### Sample Application
https://github.com/derkoe/snoop

### System information
Please provide the following information:
 - SDK Version: alr_java:3.0.4-BETA (auto injected by App Service, also tried manually with 3.1.1
 - OS type and version: Linux
 - Application Server type and version (if applicable): App Service, Spring Boot 
 - Using spring-boot? Yes
 - Additional relevant libraries (with version, if applicable): -

### Logs
[applicationinsights.log](https://github.com/microsoft/ApplicationInsights-Java/files/6858403/applicationinsights.log)
[applicationinsights-extension.log](https://github.com/microsoft/ApplicationInsights-Java/files/6858405/applicationinsights-extension.log)

### Screenshots
<img width="1053" alt="Bildschirmfoto 2021-07-21 um 21 46 20" src="https://user-images.githubusercontent.com/123199/126550830-47642f8e-69ce-42f6-8200-2aa40160616d.png">

### Additional Info
We are also running apps on Kubernetes with nginx-ingress where we get the correct address in `client_IP`

**Comment by derkoe** [comment]
It seems that the problem is with the "X-Forwarded-For" header. Azure App Service send the port and the IP address in this header - other load balancers and proxies only the IP address. 

* Azure App Service: `X-Forwarded-For: 183.13.21.23:56155`
* NGINX/HAProxy: `X-Forwarded-For: 183.13.21.23`

The agent (or App Insights itself) cannot parse the IP:Port combination and uses the App Service IP in the logs as a fallback.

---

When I add a `X-Forwarded-For` to the request to the app service this IP shows up in the logs:

```
curl -H "X-Forwarded-For: 1.2.3.4" https://poi-snoop.azurewebsites.net/
```
<img width="1049" alt="Bildschirmfoto 2021-07-21 um 22 03 22" src="https://user-images.githubusercontent.com/123199/126552726-c5fbf694-ce94-4ce1-88f5-8f0c731c79a4.png">

**Comment by trask** [comment]
@derkoe great finding! we'll target this for the next release

**Comment by trask** [comment]
Submitted fix upstream, will pull in here shortly https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3651

**Comment by trask** [comment]
this should be resolved now in [3.2.0-BETA.2](https://github.com/microsoft/ApplicationInsights-Java/releases/tag/3.2.0-BETA.2)
